app/presenters/content_item/contents_list.rb

```diff
@@ -22,5 +28,19 @@ def extract_headings_with_ids(html)
       end
       headings.compact
     end
+
+    def first_item_has_long_content?
+      first_item_content.count > WORD_COUNT_LIMIT
+    end
+
+    def first_item_content
+      element = Nokogiri::HTML(body).css('p').first
```

Comment:
I think we can optimise this a bit by only parsing the body once with Nokogiri, building the contents list as the existing implementation does, but also populating the first _n_ chapters so we can check their length.
This https://stackoverflow.com/questions/3484874/how-to-split-a-html-document-using-nokogiri does something similar by building a hash of chapters.